Bashar

iambashar opened the pull request:
Pull Request from Bashar.

iambashar committed:
ResNext added

iambashar committed:
Senet added

iambashar committed:
Update resnext.py

OCM-7898 committed:
Update train_cifar10.py

Updated for InceptionNet/GoogleNet

OCM-7898 committed:
Create googlenet.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create efficientnet.py

OCM-7898 committed:
Update efficientnet.py

Only EfficientNetB0

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create densenet.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create dla.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create convnext.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update requirements.txt

OCM-7898 committed:
Update convnext.py

OCM-7898 committed:
Create regnet.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create xception.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create dynamixer.py

OCM-7898 committed:
Update train_cifar10.py

Commit:
dpn and crossvit added

OCM-7898 committed:
Update dynamixer.py

OCM-7898 committed:
Update dynamixer.py

OCM-7898 committed:
Delete dynamixer.py

OCM-7898 committed:
Update train_cifar10.py

Commit:
pvt and pvtv2 added

iambashar committed:
HRNet added

Commit:
visformer, resmlp. vip added

iambashar committed:
Squeeze Net added

iambashar committed:
GCVit added

Commit:
version fixed

Commit:
maxvit added

iambashar committed:
Train test file refactored

Commit:
cyclemlp and asmlp added

iambashar committed:
Minor Updates in train, test, utils etc.

iambashar committed:
Minor Updates

Commit:
gfnet and msmlp added

Commit:
wavemlp and hiremlp added

OCM-7898 committed:
Create s2_mlp_v1.py

OCM-7898 committed:
Rename s2_mlp_v1.py to s2mlp_v1.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update and rename s2mlp_v1.py to s2mlpv1.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update s2mlpv1.py

OCM-7898 committed:
Update s2mlpv1.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update s2mlpv1.py

OCM-7898 committed:
Create volo.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create xcit.py

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Create raft_mlp.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update convnext.py

OCM-7898 committed:
Update train_cifar10.py

OCM-7898 committed:
Update train_cifar10.py

iambashar committed:
wandb added

niam06 commented on models/gcvit.py:
```diff
@@ -0,0 +1,828 @@
+# Copied from https://github.com/NVlabs/GCVit/blob/main/models/gc_vit.py
```
Modified from https://github.com/NVlabs/GCVit/blob/main/models/gc_vit.py to incorporate details of implementation in multiple languages, apply linting, and overall improve the readability of the code.

niam06 commented on models/gcvit.py:
```diff
@@ -0,0 +1,828 @@
+# Copied from https://github.com/NVlabs/GCVit/blob/main/models/gc_vit.py
+
+#------Original Header--------#
```
#------Extended copyright from our repository--------#
#Licensed under The MIT License [see LICENSE for details]

#------Copyright from https://github.com/NVlabs/GCVit/blob/main/models/gc_vit.py--------#

niam06 commented on models/gcvit.py:
```diff
@@ -0,0 +1,828 @@
+
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
```
Seems we cannot use this code. Let's look for an alternative.

Let's keep this as an example for us to update the main README page to motivate our work that we are modifying from works that have free to use/modify licence and that we are also using MIT License which gives users express permission to reuse code for any purpose.

niam06 commented on models/gcvit.py:
```diff
@@ -0,0 +1,828 @@
+    return x.permute(0, 3, 1, 2)
+
+
+def window_partition(x, window_size, h_w, w_w):
```
def window_partition(x: Tensor, window_size: int, h_w: int, w_w:int):
    """
    Args:
        x: Tensor, input tensor (B, H, W, C)
        window_size: int, window size for <insert short detail>
        h_w: int, <insert short detail>
        w_w: int, <insert short detail>
    Returns:
        Tensor, local window features (num_windows*B, window_size, window_size, C)
    """
# Note: not sure if x and returned variables are Tensor or Tuple(Tensor)

niam06 commented on models/gcvit.py:
```diff
@@ -0,0 +1,828 @@
+        return x
+
+
+@register_model
```
Docstring missing. Also, what is a @register model wrapper? I haven't seen this before?

iambashar committed:
train resume code fixed

iambashar committed:
learning rate issue fixed

iambashar committed:
start_epoch issue fixed

OCM-7898 committed:
Create train.py

OCM-7898 committed:
Create test.py

Taken from Bashar Branch

OCM-7898 committed:
Update utils.py

Utils file updated from Bashar branch

OCM-7898 committed:
Update __init__.py

Updated Imports

OCM-7898 committed:
Update train.py

Added new scheduler ReduceLROnPlateau

OCM-7898 committed:
Update train.py

Fixed default argument passed to scheduler

OCM-7898 committed:
Update train.py

Fixed Syntax error

OCM-7898 committed:
Update train.py

OCM-7898 committed:
Update train.py

OCM-7898 committed:
Update googlenet.py

iambashar committed:
more modularized code

iambashar committed:
schedular added

iambashar committed:
minor bug fixed

iambashar committed:
bug fixed

OCM-7898 committed:
Update __init__.py

OCM-7898 committed:
Update train.py

1. Added SGD parameters momentum and weight decay
2. Added T_max and removed args.n_epochs in CosineAnnealing
3. Modified the naming convention to rerun instead of run

OCM-7898 committed:
Update train.py

Added weight_decay and momentum to adam too

OCM-7898 committed:
Update train.py

1. Updated Adam as there is no term momentum in Adam

iambashar committed:
SGD added

iambashar committed:
small bug fixed

OCM-7898 committed:
Merge branch 'Merge_Branch' into Samia

OCM-7898 committed:
Merge pull request #2 from SamiaShashmi/Samia

Samia

iambashar commented:
Will open later with more updates.

Commit:
code refactored

OCM-7898 committed:
Delete train_cifar10.py

OCM-7898 committed:
Merge branch 'Merge_Branch' into Bashar
